tools/cldr-code/src/main/java/org/unicode/cldr/util/MemoryHelper.java

```diff
@@ -0,0 +1,63 @@
+package org.unicode.cldr.util;
+
+import java.text.CharacterIterator;
+import java.text.StringCharacterIterator;
+
+public class MemoryHelper {
+
+    /**
+     * Get the amount of memory still available for us to allocate,
+     * including not only freeMemory but also maxMemory - totalMemory
+     *
+     * https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Runtime.html#freeMemory()
+     *
+     * Generally: freeMemory <= totalMemory <= maxMemory
+     *
+     * @param callerId a string identifying the caller, used in log if verbose
+     * @param verbose if true, log the stats
+     * @return the available memory, in bytes
+     */
+    public static synchronized long availableMemory(String callerId, boolean verbose) {
+        final Runtime r = Runtime.getRuntime();
+        long freeMem = r.freeMemory();
+        long maxMem = r.maxMemory();
+        long totalMem = r.totalMemory();
+        if (freeMem > totalMem || totalMem > maxMem) {
+            log(callerId, "Values returned by Runtime violate assumptions!");
+            verbose = true;
+        }
+        long availMem = freeMem + maxMem - totalMem;
+        if (verbose) {
```

Comment:
see my TODO comment in Summary.log in this PR: "TODO: how to enable logger.info? As currently configured, it doesn't print to console" -- I just don't know how to do it

Reply:
for future

Reply:
actually if it's clear to you, whenever you can spare a few minutes i'd appreciate a quick lesson

i think what would be ideal is if we could always use a logger instead of System.out or System.err, and so that we would have the OPTION of influencing the logging level on a per-class basis

ideally anything with "err" in it should really only be for errors; and yet it should be easy to indicate when info is important -- we shouldn't be tempted to use err instead of info just because it's less likely to be hidden